### DISCUSSION TITLE
Dynamically sized HDD images for VMs using Apple Virtualization?

### DISCUSSION THREAD

**Discussion by alshdavid** [utmapp/UTM#4602] [UTM Help (macOS)] [answered]
I might be missing something but the Hard Drive images created using UTM with Apple Virtualization pre-allocate the entire disk image size upon creation.

Is there a way to have them dynamically sized based on their contents?

![image](https://user-images.githubusercontent.com/12656294/199646075-41654693-63d0-4f6d-8942-7123560d200f.png)

**Comment by osy** [comment, marked as answer]
If you are using APFS then it actually doesn’t use that space. Right click more info and you’ll see the actual size it’s occupying.
